client_http.hpp

```diff
@@ -16,22 +16,29 @@ namespace SimpleWeb {
   using errc = std::errc;
   using system_error = std::system_error;
   namespace make_error_code = std;
-  using string_view = const std::string &; // TODO c++17: use std::string_view
 } // namespace SimpleWeb
 #else
 #include <boost/asio.hpp>
 #include <boost/asio/steady_timer.hpp>
-#include <boost/utility/string_ref.hpp>
 namespace SimpleWeb {
   namespace asio = boost::asio;
   using error_code = boost::system::error_code;
   namespace errc = boost::system::errc;
   using system_error = boost::system::system_error;
   namespace make_error_code = boost::system::errc;
-  using string_view = boost::string_ref;
 } // namespace SimpleWeb
 #endif
 
+#if defined(__has_include) && __has_include(<string_view>) && (__cplusplus >= 201402L || (defined(_MSC_VER) && _MSC_VER >= 1910))
```

Comment:
Is the __has_include checks needed if one checks for the __cplusplus standard? Also, what is up with MSVC, do they not support the standardised __cplusplus define?

According to https://gcc.gnu.org/onlinedocs/cpp/Standard-Predefined-Macros.html it should maybe be `>= 201703L`?

Reply:
They _DO_ support it, but don't feel they support enough C++17 to move the `__cplusplus` version up from like 99. They have been using this excuse forever, that until they are 100% compliant (and noone is) they won't bump it. So, I added secondary check for `_MSC_VER` to make sure it was a new enough version to support it. The `__has_include` is simply insurance, just in case they don't have a version of the SDK that supports it.... but it could potentially be skipped, your call 😃

Reply:
Even more annoying is that MSVC 15.5+ is actually fairly C++17 complete, and is definitely C++11 complete, so I'm not sure why they haven't even bumped the define up to C++11.

Reply:
Ah, ok, sounds good then:) Maybe remove __has_include for now and add it later if needed. But it should be `>= 201703L `.

Reply:
Should be a question mark at the end there:)